### PR TITLE
Call rgxp callback in BackendPassword controller

### PR DESCRIPTION
**Pull request by richardhj**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes terminal42/contao-password-validation#3
| Docs PR or issue | 

The `BackendPassword` controller implements its own form validation.
Usually, we would do a `$objWidget = new Password();` and call `$objWidget->validate()` to have all form validation done, but this is not the case in the `BackendPassword` controller.

In this case, the BackendPassword controller lacks the `rgxp` check, which validates the value if a custom rgxp is set on the password field.
This results in an issue in our "terminal42/contao-password-validation" extension that validates the password.

I don't know, why the `BackendPassword` controller is not using the Widget to build and validate the form, but there might be a reason. I decided to check for registered rgxp hooks.

## Thoughts
- [ ] In the back end scope, you usually can throw \Exceptions in the save_callback to add an error to the widget. This will not work in the current implementation. Do we need to change this as well to be consistent?
- [ ] Do we rather need to change the whole implementation, e.g. populate  a `$objWidget = new Password();`? 

cc @aschempp

### CODE REVIEW

File: core-bundle/src/Resources/contao/controllers/BackendPassword.php [hunk 1]
```diff
@@ -81,21 +81,46 @@ public function run()
 			// Save the data
 			else
 			{
+				$this->loadDataContainer('tl_user');
+
+				$dc = new \DC_Table('tl_user');
+				$dc->id = $this->User->id;
+
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/controllers/BackendPassword.php:88]: I would prefer to keep this code in the `elseif` and `else` condition, even though this means duplicating it. Loading a DCA and instantiating a new DC object are expensive operations that are not necessary for the `if` condition.

**Review by leofeyer — CHANGES_REQUESTED** [review]

**Comment by leofeyer** [comment]
> Do we rather need to change the whole implementation, e.g. populate  a `$objWidget = new Password();`?

Yes, I think so.

**Comment by aschempp** [comment]
I think so too. Maybe the profile tab/module might give us a hint on how to implement that. As far as I remember, thats some special backend view as well.

**Comment by aschempp** [comment]
@richardhj can you take care of this?

**Comment by leofeyer** [comment]
@richardhj Since Contao 4.4 is now in the "security updates only" phase of the LTS cycle, please rebase this PR on the `4.9` branch.

**Comment by AlexanderWillner** [comment]
Would you merge it if I rebase it for 4.13?

**Pull request closed without merging**